bokeh/models/widgets/inputs.py

```diff
@@ -97,6 +98,37 @@ class TextInput(InputWidget):
     """)
 
 
+class TextAreaInput(InputWidget):
+    ''' Multi-line input widget.
+
+    '''
+
+    value = String(default="", help="""
+    Initial or entered text value.
+    """)
+
+    callback = Instance(Callback, help="""
+    A callback to run in the browser whenever the user unfocuses the TextInput
+    widget by hitting Enter or clicking outside of the text box area.
+    """)
+
+    placeholder = String(default="", help="""
+    Placeholder for empty input field
+    """)
```

Comment:
@hhuuggoo it looks like `value`, `callback` and `placeholder` are shared with `TextInput`, what do you think about factoring into a `TextInputBase` or similar? I think it would be fine if this is just on the Python side, but it could also be done on the JS side for a small reduction in code size.

Reply:
sure I can do that - I'm also not sure about callback, I saw some initial discussions about removing it but I left it in cause TextInput still has it.

Reply:
Oh that's a good point, I would say it should be omitted in new classes, and many be deprecated for old ones at some future point (no set plans tho). IN that case the base could just factor `value and `placeholder`